Automattic/wp-calypso
# Stats: Support dismissible Tier Upgrade notice with the API

Related to #85122 

## Proposed Changes

* Update the Tier Upgrade notice id to `tier_upgrade`.
* Apply the notice mutation to `TierUpgradeNotice` for 7 days postponement.

## Testing Instructions



* Apply the Diff D132557-code and sandbox public API.
* Spin this change on local Calypso.
* Edit code to show the Tier Upgrade notice:
https://github.com/Automattic/wp-calypso/blob/eaf4f1ace57675c73b1110467b809220560f7930/client/my-sites/stats/stats-notices/tier-upgrade-notice.tsx#L25-L26
* Click the close button on the top right of the Tier Upgrade notice banner.
* Ensure the notice disappears after dismissal.
* Reload the page.
* Ensure the `tier_upgrade` of the notice API endpoint response is `false`.

<img width="925" alt="截圖 2023-12-20 上午12 45 12" src="https://github.com/Automattic/wp-calypso/assets/6869813/0b239603-4914-49a4-bca1-e4a0339ae6f2">

* Ensure the notice does not show for 7 days.

## Pre-merge Checklist
